src/main/java/com/starkbank/BoletoHolmes.java

```diff
@@ -0,0 +1,384 @@
+     * updated [string, default null]: latest update datetime for the holmes. ex: datetime.datetime(2020, 3, 10, 10, 30, 0, 0)
+     */
+
+    public BoletoHolmes(Map<String, Object> data) throws Exception {
+        super(null);
+        HashMap<String, Object> dataCopy = new HashMap<>(data);
+
+        this.tags = (String[]) dataCopy.remove("tags");
+        this.boletoId = (String) dataCopy.remove("boletoId");
+        this.status = (String) dataCopy.remove("status");
+        this.result = (String) dataCopy.remove("result");
+        this.created = (String) dataCopy.remove("created");
+        this.updated = (String) dataCopy.remove("updated");
+
+        if (!dataCopy.isEmpty()){
+            throw new Exception("Unknown parameters used in constructor: [" + String.join(", ", dataCopy.keySet()) + "]");
+        }
+    }
+
+    /**
+     * Retrieve a specific BoletoHolmes
+     * <p>
+     * Receive a single BoletoHolmes object previously created by the Stark Bank API by passing its id
+     * <p>
+     * Parameters (required):
+     * @param id [string]: object unique id. ex: "5656565656565656"
+     * Return:
+     * @return BoletoHolmes object with updated attributes
+     * @throws Exception error in the request 
+     */
+    public static BoletoHolmes get(String id) throws Exception {
+        return BoletoHolmes.get(id, null);
+    }
+
+    /**
+     * Retrieve a specific BoletoHolmes
+     * <p>
+     * Receive a single BoletoHolmes object previously created by the Stark Bank API by passing its id
+     * <p>
+     * Parameters (required):
+     * @param id [string]: object unique id. ex: "5656565656565656"
+     * <p>
+     * Parameters (optional):
+     * @param user [Project object]: Project object. Not necessary if starkbank.user was set before function call
+     * Return:
+     * @return BoletoHolmes object with updated attributes
+     * @throws Exception error in the request 
+     */
+    public static BoletoHolmes get(String id, Project user) throws Exception {
+        return Rest.getId(data, id, user);
+    }
+
+    /**
+     * Create BoletoHolmes
+     * <p>
+     * Send a list of BoletoHolmes objects for creation in the Stark Bank API
+     * <p>
+     * Parameters (required):
+     * @param holmes [list of BoletoHolmes objects]: list of BoletoHolmes objects to be created in the API
+     * <p>
+     * Parameters (optional):
+     * @param user [Project object]: Project object. Not necessary if starkbank.user was set before function call
+     * <p>
+     * Return:
+     * @return list of BoletoHolmes objects with updated attributes
+     * @throws Exception error in the request
+     */
+    @SuppressWarnings("unchecked")
+    public static List<BoletoHolmes> create(List<?> holmes, Project user) throws Exception {
+        List<BoletoHolmes> holmesList = new ArrayList<>();
+        for (Object sherlock : holmes){
+            if (sherlock.getClass() == HashMap.class){
```

Comment:
Em Java, mapas são uma interface. O usuário poderia ter criado um TreeMap. A verificação ideal seria if(sherlock instanceof Map<?,?>)

Reply:
instanceof tests whether the object reference on the left-hand side (LHS) is an instance of the type on the right-hand side (RHS) or some subtype.

getClass() == ... tests whether the types are identical.

Reply:
Faz sentido mesmo. Para comportar um TreeMap teria que mudar mesmo o método 'create' das features. Do jeito que tá só os dois tipos: lista de BoletoHolmes ou HashMap são aceitáveis. 
Acho que não faz mal. Vou lançar um commit com um fix disso.

Reply:
Faz sentido mesmo. Para comportar um TreeMap teria que mudar mesmo o método 'create' das features. Do jeito que tá só dois tipos são aceitáveis: o tipo do objeto em questão ou HashMap
Acho que não faz mal. Vou lançar um commit com um fix disso.

Reply:
Faz sentido mesmo. Para comportar um TreeMap teria que mudar mesmo o método 'create' das features. Do jeito que tá só dois tipos são aceitáveis: o tipo do objeto em questão ou HashMap.
Acho que não faz mal. Vou lançar um commit com um fix disso.